### PR TITLE
Pretty show

### LINKED ISSUE #43: Make show return "toList [1, 2, 3]" instead of "(Cons 1 (Cons 2 (Cons 3 Nil)))"
I think it would be beneficial to make `show` return `"toList [1, 2, 3]"` instead of `"(Cons 1 (Cons 2 (Cons 3 Nil)))"`, as that's much more readable.

### PR DESCRIPTION
**Pull request by rightfold**
Fixes #43. Supersedes #46.

### CODE REVIEW

File: src/Data/List/Lazy.purs [hunk 1]
```diff
@@ -643,14 +645,26 @@ transpose xs =
           (x : mapMaybe head xss) : transpose (xs : mapMaybe tail xss)
 
 --------------------------------------------------------------------------------
+-- Infinite list inspection ----------------------------------------------------
+--------------------------------------------------------------------------------
+
+-- | Like `show`, but only shows the first n elements. Useful for potentially
+-- | infinite lists.
+showPrefix :: forall a. (Show a) => Int -> List a -> String
```
Comment (paf31) [inline src/Data/List/Lazy.purs]: I think if users need this, they can just write `show <<< take n`. I do this frequently in Haskell, for example.

**Comment by paf31** [comment]
👍

File: src/Data/List/Lazy.purs (hunk 1 above)
Comment (rightfold) [inline src/Data/List/Lazy.purs]: Fair enough, I'll rebase it out

**Comment by garyb** [comment]
I just merged #74 which had an alternative pretty-`show` using `:`, so closing this. This would have been equally good in my book, so sorry it never went in!

**Pull request closed without merging**